File: packages/shared/src/super-domains.ts
export interface SuperDomain {
  id: string
  label: string  // 영어 fallback. i18n key 는 home.domain.<id>
  emoji: string
  tags: string[]
  paletteVar: string
}

export const SUPER_DOMAINS: SuperDomain[] = [
  { id: 'mobile',   label: 'Mobile',   emoji: '📱',
    tags: ['mobile'],                                  paletteVar: '--graph-domain-mobile' },
  { id: 'software', label: 'Software', emoji: '💻',
    tags: ['software', 'programming', 'architecture'], paletteVar: '--graph-domain-software' },
  { id: 'hardware', label: 'Hardware', emoji: '🔧',
    tags: ['semiconductor', 'electronics', 'display'], paletteVar: '--graph-domain-hardware' },
  { id: 'telecom',  label: 'Telecom',  emoji: '📡',
    tags: ['telecom'],                                 paletteVar: '--graph-domain-telecom' },
]

export const NOISE_TAGS = new Set([
  '템플릿', '미팅', 'faq', 'intro', 'sample', 'namu-archive', 'imported-bulk',
])

File: apps/api/app/lib/super_domains.py
"""Mirror of packages/shared/src/super-domains.ts — keep in sync.

If you change SUPER_DOMAINS in TS, update this file too.
"""
from dataclasses import dataclass

@dataclass(frozen=True)
class SuperDomain:
    id: str
    label: str
    emoji: str
    tags: tuple[str, ...]
    palette_var: str

SUPER_DOMAINS: list[SuperDomain] = [
    SuperDomain('mobile',   'Mobile',   '📱', ('mobile',),                                  '--graph-domain-mobile'),
    SuperDomain('software', 'Software', '💻', ('software', 'programming', 'architecture'),  '--graph-domain-software'),
    SuperDomain('hardware', 'Hardware', '🔧', ('semiconductor', 'electronics', 'display'),  '--graph-domain-hardware'),
    SuperDomain('telecom',  'Telecom',  '📡', ('telecom',),                                 '--graph-domain-telecom'),
]

NOISE_TAGS: frozenset[str] = frozenset({
    '템플릿', '미팅', 'faq', 'intro', 'sample', 'namu-archive', 'imported-bulk',
})

def by_id(domain_id: str) -> SuperDomain | None:
    return next((d for d in SUPER_DOMAINS if d.id == domain_id), None)
